src/tunnel.cc

```diff
@@ -625,6 +661,20 @@ TunnelStateData::Connection::write(const char *b, int size, AsyncCall::Pointer &
     Comm::Write(conn, b, size, callback, free_func);
 }
 
+void
+TunnelStateData::Connection::setCloseHandler(CLCB *handler, TunnelStateData *tunnelState)
```

Comment:
If it does not cause a lot of trouble, please use an AsyncCall instead of raw CLCB. Comm will convert CLCB into an AsyncCall for us, but that conversion cannot know the handler name.